armi/utils/__init__.py

```diff
@@ -806,17 +806,38 @@ def merge(self, *otherDictionaries) -> None:
 
 def safeCopy(src: str, dst: str) -> None:
     """This copy overwrites ``shutil.copy`` and checks that copy operation is truly completed before continuing."""
-    waitTime = 0.01  # 10 ms
+    # Convert files to OS-independence
+    src = os.path.abspath(src)
+    dst = os.path.abspath(dst)
     if os.path.isdir(dst):
         dst = os.path.join(dst, os.path.basename(src))
     srcSize = os.path.getsize(src)
-    shutil.copyfile(src, dst)
```

Comment:
Can we wrap this in a try/except and then proceed with the Windows / Linux check? Additionally, the detection of Windows / Linux and any language specific utility code requirements would be helpful at the start on ARMI. Is it possible to search the code base for these (maybe in a couple places where we use if "Win..." or if "Linux" and then pull those checks to ARMI __init__ or bootstrap? Just thinking about a potential speed up for failure if something is missing

Reply:
I don't like keeping code in that is that fragile. I think the change I made is a little more robust

As far as the OS-check, I agree we shouldn't be adding a gazillion checks throughout the code. We should catch the OS failure during init or bootstrap like you suggest, then we can safely do "if win this else that" throughout the code.

But @john-science gets the final call there.

Reply:
Actually @jakehader: I can have the windows `if` do the old/shutil approach, and the linux `if` do the `cp`/command line stuff. I'm fine with that. We know it works on windows and has for years.

Reply:
@jakehader for the OS check, we are going to handle that in a different ticket. The code for now should stay until that ticket is addressed. It's out of scope for this PR.

And for windows, @john-science prefers the `copy` approach over the `shutil` approach. I don't care which we go with, but it seems cleaner to me to have parallel approaches implemented, so I lean towards the way I have it coded now. 

So as of right now John is just awaiting me to complete some internal testing to prove this PR is ready.

Reply:
@opotowsky @jakehader That's a nice idea, I like it.  But I decided to make that it's own [ticket](https://github.com/terrapower/armi/issues/1693).

This is for Arrielle's expediency right now.  But more so, I like to keep "one feature = one PR" when I can.

Reply:
HAHA @john-science jinx
https://github.com/terrapower/armi/issues/1694

Reply:
feel free to close my duplicate ticket